tests/zfs-tests/tests/functional/rsend/send_freeobjects.ksh

```diff
@@ -0,0 +1,74 @@
+#!/bin/ksh
+
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright (c) 2017 by Lawrence Livermore National Security, LLC.
+#
+
+. $STF_SUITE/include/libtest.shlib
+. $STF_SUITE/tests/functional/rsend/rsend.kshlib
+
+#
+# Description:
+# Verify FREEOBJECTS record frees sequential objects (See
+# https://github.com/zfsonlinux/zfs/issues/6694)
+#
+# Strategy:
+# 1. Create three files with sequential object numbers, f1 f2 and f3
+# 2. Delete f2
+# 3. Take snapshot A
+# 4. Delete f3
+# 5. Take snapshot B
+# 6. Receive a full send of A
+# 7. Receive an incremental send of B
+# 8. Fail test if f3 exists on received snapshot B
+
+verify_runnable "both"
+
+log_assert "Verify FREEOBJECTS record frees sequential objects"
+
+sendds=sendfo
+recvds=recvfo
+f1=/$POOL/$sendds/f1
+f2=/$POOL/$sendds/f2
+f3=/$POOL/$sendds/f3
+
+log_must zfs create $POOL/$sendds
+
+tries=100
+for ((i=0; i<$tries; i++)); do
```

Comment:
Actually, it looks like this did happen in the last test run.

http://build.zfsonlinux.org/builders/CentOS%206%20x86_64%20%28TEST%29/builds/489/steps/shell_8/logs/stdio

```
Test: /usr/share/zfs/zfs-tests/tests/functional/rsend/send_freeobjects (run as root) [10:00] [KILLED]
19:19:53.38 ASSERTION: Verify FREEOBJECTS record frees sequential objects
19:19:53.42 SUCCESS: zfs create testpool/sendfo
```

Reply:
Sigh. Of course it almost always exits the loop on the first or second iteration in manual testing. I may have to push a more verbose version to debug the builders.

Reply:
Turns out SELinux xattrs were consuming the intervening object numbers. Updated test case to force `xattr=sa` to work around this.